app/controllers/home_controller.rb

```diff
@@ -1,9 +1,11 @@
 class HomeController < ApplicationController
   def index
     @posts = Post.where(status: Post.statuses[:accepted])
-                 .order(created_at: :desc)
-                 .includes({ user: { avatar_attachment: :blob } }, :tags, :cover_image_attachment)
+                 .includes({ user: { avatar_attachment: :blob } }, :tags)
     @posts = Posts::PostsFilterService.call(@posts, params)
+    @posts = Posts::PostsSortService.call(@posts, params)
```

Comment:
Should write in one service only, because filter or sort related to a business logic, you can wrap it up in a service call like ListPostsService, will be easier to maintain and refactor